Tests/Common/TimeTests.cs

```diff
@@ -76,11 +76,48 @@ public void GetStartTimeForTradeBarsHandlesWeekends()
             var expectedStart = new DateTime(2015, 08, 21);
             var barSize = TimeSpan.FromDays(1);
             var hours = SecurityExchangeHoursTests.CreateUsEquitySecurityExchangeHours();
-            var start = Time.GetStartTimeForTradeBars(hours, end, barSize, 7, false);
+            var start = Time.GetStartTimeForTradeBars(hours, end, barSize, 7, false, hours.TimeZone);
             // from noon, back up to 9am (3 hours) then skip night, so from 4pm, back up to noon, 4 more hours
             Assert.AreEqual(expectedStart, start);
         }
 
+        public static IEnumerable<TestCaseData> ForexHistoryDates => new List<TestCaseData>
```

Comment:
nit code style - we usually place these static test cases at the bottom, do they need to be public?

Reply:
done